libraries/plugins/account_history/account_history_plugin.cpp

```diff
@@ -99,7 +111,7 @@ void account_history_plugin_impl::update_account_histories( const signed_block&
       {
          // Note: the 2nd and 3rd checks above are for better performance, when the db is not clean,
          //       they will break consistency of account_stats.total_ops and removed_ops and most_recent_op
-         _oho_index->use_next_id();
+         skip_oho_id();
```

Comment:
@ronakpatel3110  - why this change ? ideally changes like this should be explained via comments.

Reply:
When creating new objects, old_index_next_ids will be updated in on_create(), when undoing, the value will be reverted to saved old id.
But in history plugin we didn't save old ids in old_index_next_ids, thus no reversion will be done when undoing.
To fix this, a new lambda function is created with name `skip_oho_id` which will check if undo happened then reset the id else use the next id.

A comment is already added here in lambda function: